## DATA ANALYSIS WITH PANDAS

Importamos librerías

In [2]:
import pandas as pd
import numpy as np

Restaurant data with consumer rating

https://www.kaggle.com/uciml/restaurant-data-with-consumer-ratings

Elegimos un dataset sobre restaurantes, rating y perfil de usuario en México. Nuestra hipótesis principal es que queremos invertir en un nuevo restaurante en México y queremos conocer cuáles son los restaurantes mejores valorados y el tipo de usuario. 

Importamos el dataset para poder analizar. Concretamente este dataset tiene varios dataset relacionados: restaurantes en méxico, tipo de cocina de los restaurantes, el rating de cada restaurante por usuairo y el perfil de usuario.  


### Restaurants
El primer dataset que tenemos es restaurants, tenemos que limpiarlo porque hay ciertos datos que no nos interesan. 

In [3]:
#Importamos el dataset.
restaurants = pd.read_csv('geoplaces2.csv')
display(restaurants.head())
print(restaurants.columns)


placeID   latitude   longitude  \
0   134999  18.915421  -99.184871   
1   132825  22.147392 -100.983092   
2   135106  22.149709 -100.976093   
3   132667  23.752697  -99.163359   
4   132613  23.752903  -99.165076   

                                      the_geom_meter  \
0  0101000020957F000088568DE356715AC138C0A525FC46...   
1  0101000020957F00001AD016568C4858C1243261274BA5...   
2  0101000020957F0000649D6F21634858C119AE9BF528A3...   
3  0101000020957F00005D67BCDDED8157C1222A2DC8D84D...   
4  0101000020957F00008EBA2D06DC8157C194E03B7B504E...   

                             name                                 address  \
0                 Kiku Cuernavaca                              Revolucion   
1                 puesto de tacos  esquina santos degollado y leon guzman   
2      El Rinc�n de San Francisco                         Universidad 169   
3  little pizza Emilio Portes Gil                 calle emilio portes gil   
4                   carnitas_mata                  lic. Emilio portes gil   

              city            state country fax      ...        \
0       Cuernavaca          Morelos  Mexico   ?      ...         
1           s.l.p.           s.l.p.  mexico   ?      ...         
2  San Luis Potosi  San Luis Potosi  Mexico   ?      ...         
3        victoria        tamaulipas       ?   ?      ...         
4         victoria       Tamaulipas  Mexico   ?      ...         

             alcohol smoking_area dress_code     accessibility   price  \
0  No_Alcohol_Served         none   informal  no_accessibility  medium   
1  No_Alcohol_Served         none   informal        completely     low   
2          Wine-Beer  only at bar   informal         partially  medium   
3  No_Alcohol_Served         none   informal        completely     low   
4  No_Alcohol_Served    permitted   informal        completely  medium   

                     url Rambience franchise    area other_services  
0  kikucuernavaca.com.mx  familiar         f  closed           none  
1                      ?  familiar         f    open           none  
2                      ?  familiar         f    open           none  
3                      ?  familiar         t  closed           none  
4                      ?  familiar         t  closed           none  

[5 rows x 21 columns]

Index(['placeID', 'latitude', 'longitude', 'the_geom_meter', 'name', 'address',
       'city', 'state', 'country', 'fax', 'zip', 'alcohol', 'smoking_area',
       'dress_code', 'accessibility', 'price', 'url', 'Rambience', 'franchise',
       'area', 'other_services'],
      dtype='object')


In [4]:
restaurants.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130 entries, 0 to 129
Data columns (total 21 columns):
placeID           130 non-null int64
latitude          130 non-null float64
longitude         130 non-null float64
the_geom_meter    130 non-null object
name              130 non-null object
address           130 non-null object
city              130 non-null object
state             130 non-null object
country           130 non-null object
fax               130 non-null object
zip               130 non-null object
alcohol           130 non-null object
smoking_area      130 non-null object
dress_code        130 non-null object
accessibility     130 non-null object
price             130 non-null object
url               130 non-null object
Rambience         130 non-null object
franchise         130 non-null object
area              130 non-null object
other_services    130 non-null object
dtypes: float64(2), int64(1), object(18)
memory usage: 21.4+ KB


In [5]:
#Eliminamos columnas que no nos interesan para el análisis. 
restaurants = restaurants.drop(restaurants.columns[[1,2,3,5,7,9,10,16]], axis=1)


In [71]:
# Modificamos datos
restaurants = restaurants[restaurants['country'] == restaurants['country'].str.replace('?', 'Mexico')]
restaurants

placeID                                               name  \
0     134999                                    Kiku Cuernavaca   
1     132825                                    puesto de tacos   
2     135106                         El Rinc�n de San Francisco   
4     132613                                      carnitas_mata   
5     135040                           Restaurant los Compadres   
6     132732                                 Taqueria EL amigo    
8     132609                           Pollo_Frito_Buenos_Aires   
9     135082                               la Estrella de Dimas   
10    135070                                     Restaurante 75   
11    135069                          Abondance Restaurante Bar   
12    135065                               El angel Restaurante   
13    135076                          Restaurante Pueblo Bonito   
14    135086                        Mcdonalds Parque Tangamanga   
15    132870                     Tortas y hamburguesas el gordo   
16    132854                                            Sirlone   
17    132937                                        rockabilly    
18    132856                                      Unicols Pizza   
19    132668                                     TACOS EL GUERO   
20    135033                   Restaurant El Muladar de Calzada   
21    132862                               La Posada del Virrey   
22    134983  Restaurant and Bar and Clothesline Carlos N Ch...   
23    132851                                                KFC   
24    135013                                          Giovannis   
25    135042                        Restaurant Oriental Express   
26    132768                                 Mariscos Tia Licha   
28    135045                            Restaurante la Gran Via   
29    132847                                       don burguers   
30    135054                   Restaurante y Pescaderia Tampico   
32    135053              La Fontana Pizza Restaurante and Cafe   
33    135050                    Restaurante la Estrella de Dima   
..       ...                                                ...   
85    132885                                  Hamburguesas saul   
86    132740                      Carreton de Flautas y Migadas   
89    132706                                 Gorditas Dona Tota   
91    134987                                  Chilis Cuernavaca   
92    132572                                       Cafe Chaires   
93    132958                                 tacos los volcanes   
94    132733                                      Little Cesarz   
97    135027                               Restaurant Orizatlan   
99    135039                    Restaurant de Mariscos de Picon   
100   134996                               Sanborns Casa Piedra   
101   132755                               La Estrella de Dimas   
102   135043                                      pizza clasica   
104   135049                Restaurante de Mariscos la Langosta   
105   135073                       Restaurante Bar El Gallinero   
106   132560                                 puesto de gorditas   
108   135075                               Mariscos El Pescador   
112   134986                           Restaurant Las Mananitas   
113   135062                      Restaurante El Cielo Potosino   
114   132766                                             Mikasa   
115   135088                                  Cafeteria cenidet   
116   135058                               Restaurante Tiberius   
117   132754                                    Cabana Huasteca   
119   135032                 Cafeteria y Restaurant El Pacifico   
120   135028                                        La Virreina   
121   135085                             Tortas Locas Hipocampo   
122   132663                                          tacos abi   
124   132583                                   McDonalds Centro   
125   132866                                        

In [7]:
#Comprobamos valores nulos
restaurants.isnull()

placeID   name   city  country  alcohol  smoking_area  dress_code  \
0      False  False  False    False    False         False       False   
1      False  False  False    False    False         False       False   
2      False  False  False    False    False         False       False   
4      False  False  False    False    False         False       False   
5      False  False  False    False    False         False       False   
6      False  False  False    False    False         False       False   
8      False  False  False    False    False         False       False   
9      False  False  False    False    False         False       False   
10     False  False  False    False    False         False       False   
11     False  False  False    False    False         False       False   
12     False  False  False    False    False         False       False   
13     False  False  False    False    False         False       False   
14     False  False  False    False    False         False       False   
15     False  False  False    False    False         False       False   
16     False  False  False    False    False         False       False   
17     False  False  False    False    False         False       False   
18     False  False  False    False    False         False       False   
19     False  False  False    False    False         False       False   
20     False  False  False    False    False         False       False   
21     False  False  False    False    False         False       False   
22     False  False  False    False    False         False       False   
23     False  False  False    False    False         False       False   
24     False  False  False    False    False         False       False   
25     False  False  False    False    False         False       False   
26     False  False  False    False    False         False       False   
28     False  False  False    False    False         False       False   
29     False  False  False    False    False         False       False   
30     False  False  False    False    False         False       False   
32     False  False  False    False    False         False       False   
33     False  False  False    False    False         False       False   
..       ...    ...    ...      ...      ...           ...         ...   
85     False  False  False    False    False         False       False   
86     False  False  False    False    False         False       False   
89     False  False  False    False    False         False       False   
91     False  False  False    False    False         False       False   
92     False  False  False    False    False         False       False   
93     False  False  False    False    False         False       False   
94     False  False  False    False    False         False       False   
97     False  False  False    False    False         False       False   
99     False  False  False    False    False         False       False   
100    False  False  False    False    False         False       False   
101    False  False  False    False    False         False       False   
102    False  False  False    False    False         False       False   
104    False  False  False    False    False         False       False   
105    False  False  False    False    False         False       False   
106    False  False  False    False    False         False       False   
108    False  False  False    False    False         False       False   
112    False  False  False    False    False         False       False   
113    False  False  False    False    False         False       False   
114    False  False  False    False    False         False       False   
115    False  False  False    False    False         False       False   
116    False  False  False    False    False         False       False   
117    False  False  False    False    False         False       False   
119    Fa

### Restaurants & Cuisine 
El segundo dataset que queremos limpiar y modificar para obetener los datos que queremos es los tipos de comida de cada restaurante. 
Para ello debemos de limpiar las categorías y renombrar categorías más globales, una por cada restaurante. 

In [8]:
# Exportamos dataset
restaurant_type = pd.read_csv('chefmozcuisine.csv')
display(restaurant_type.head())



placeID        Rcuisine
0   135110         Spanish
1   135109         Italian
2   135107  Latin_American
3   135106         Mexican
4   135105       Fast_Food

In [140]:
restaurant_type.groupby('placeID')

AttributeError: Cannot access callable attribute 'set_index' of 'DataFrameGroupBy' objects, try using the 'apply' method

In [131]:
restaurant_type.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 916 entries, 0 to 915
Data columns (total 2 columns):
placeID     916 non-null int64
Rcuisine    916 non-null category
dtypes: category(1), int64(1)
memory usage: 11.1 KB


In [9]:
# Modificamos el tipo de objeto de Rcuisine a Categórico. 
restaurant_type[restaurant_type.select_dtypes(['object']).columns] = restaurant_type.select_dtypes(['object']).apply(lambda x: x.astype('category'))


In [10]:
#Vemos todas la categorías.
restaurant_type['Rcuisine'].cat.categories

Index(['Afghan', 'African', 'American', 'Armenian', 'Asian', 'Bagels',
       'Bakery', 'Bar', 'Bar_Pub_Brewery', 'Barbecue', 'Brazilian',
       'Breakfast-Brunch', 'Burgers', 'Cafe-Coffee_Shop', 'Cafeteria',
       'California', 'Caribbean', 'Chinese', 'Contemporary',
       'Continental-European', 'Deli-Sandwiches', 'Dessert-Ice_Cream', 'Diner',
       'Dutch-Belgian', 'Eastern_European', 'Ethiopian', 'Family', 'Fast_Food',
       'Fine_Dining', 'French', 'Game', 'German', 'Greek', 'Hot_Dogs',
       'International', 'Italian', 'Japanese', 'Juice', 'Korean',
       'Latin_American', 'Mediterranean', 'Mexican', 'Mongolian',
       'Organic-Healthy', 'Persian', 'Pizzeria', 'Polish', 'Regional',
       'Seafood', 'Soup', 'Southern', 'Southwestern', 'Spanish', 'Steaks',
       'Sushi', 'Thai', 'Turkish', 'Vegetarian', 'Vietnamese'],
      dtype='object')

In [61]:
#Definimos una función para concatenar los valores de Rcuisine
def Rcousine (df):
    return (' , '.join(df['Rcuisine']))

In [72]:
# Aplicamos la función a nuestro dataframe y renombramos la columnas
restaurant_type_agg = pd.DataFrame(restaurant_type.groupby('placeID', as_index=True).apply(Rcousine))
restaurant_type_agg.columns = ['Food_type']
restaurant_type_agg.head()

Food_type
placeID                  
132001      Dutch-Belgian
132002            Seafood
132003      International
132004            Seafood
132005   French , Seafood

In [82]:
#Podemos unir el dataset de restaurantes con el dataset de tipos de comida
rest_new = pd.merge(restaurants, restaurant_type_agg, on='placeID')
rest_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 76 entries, 0 to 75
Data columns (total 14 columns):
placeID           76 non-null int64
name              76 non-null object
city              76 non-null object
country           76 non-null object
alcohol           76 non-null object
smoking_area      76 non-null object
dress_code        76 non-null object
accessibility     76 non-null object
price             76 non-null object
Rambience         76 non-null object
franchise         76 non-null object
area              76 non-null object
other_services    76 non-null object
Food_type         76 non-null object
dtypes: int64(1), object(13)
memory usage: 8.9+ KB


Comprobamos que rest_new, nuestro nuevo dataframe donde tenemos toda la información del dataset de "restaurants" más toda la info del tipo de comida, solo nos devuelve 76 entradas. Podemos concluir que solo 76 de los 130 restaurantes que tenemos tiene información sobre el tipo de comida. 
Podríamos realizar un segundo método de unión que nos llevaría mucho tiempo limpiar los nulos y cambiar el tipo de datos NaN de la columna "food_type". De momento vamos a dejarlo así y no utilizar la columna de food_type. 

    ```pd.concat([restaurants, restaurant_type_agg], join = 'outer')
   

### User Rating
Tenemos un tercer dataframe con información sobre el rating de restaurantes. 

In [108]:
user_rating = pd.read_csv('rating_final.csv')
user_rating


userID  placeID  rating  food_rating  service_rating
0     U1077   135085       2            2               2
1     U1077   135038       2            2               1
2     U1077   132825       2            2               2
3     U1077   135060       1            2               2
4     U1068   135104       1            1               2
5     U1068   132740       0            0               0
6     U1068   132663       1            1               1
7     U1068   132732       0            0               0
8     U1068   132630       1            1               1
9     U1067   132584       2            2               2
10    U1067   132733       1            1               1
11    U1067   132732       1            2               2
12    U1067   132630       1            0               1
13    U1067   135104       0            0               0
14    U1067   132560       1            0               0
15    U1103   132584       1            2               1
16    U1103   132732       0            0               2
17    U1103   132630       1            2               0
18    U1103   132613       2            2               2
19    U1103   132667       1            2               2
20    U1103   135104       1            2               0
21    U1103   132663       1            0               2
22    U1103   132733       2            2               2
23    U1107   132660       2            2               1
24    U1107   132584       2            2               2
25    U1107   132733       2            2               2
26    U1044   135088       2            2               2
27    U1044   132583       1            2               1
28    U1070   132608       2            2               1
29    U1070   132609       1            1               1
...     ...      ...     ...          ...             ...
1131  U1087   132732       2            2               1
1132  U1087   132740       1            2               1
1133  U1087   132560       1            2               1
1134  U1087   132594       1            2               1
1135  U1087   132663       1            2               1
1136  U1065   132626       1            0               1
1137  U1065   132706       2            1               2
1138  U1129   132561       1            2               2
1139  U1065   132665       1            1               1
1140  U1129   132665       0            0               0
1141  U1065   132561       1            1               1
1142  U1039   132626       2            2               1
1143  U1039   132564       1            1               2
1144  U1065   132654       1            1               1
1145  U1039   132665       1            1               2
1146  U1130   132654       0            0               0
1147  U1130   132561       0            0               0
1148  U1130   132626       0            0               0
1149  U1130   132706       1            2               1
1150  U1043   132608       1            1               1
1151  U1043   132609       1            2               1
1152  U1011   132717       2            2               1
1153  U1043   132613       1            1               1
1154  U1011   132564       1            1               1
1155  U1043   132732       1            1               1
1156  U1043   132630       1            1               1
1157  U1011   132715       1            1               0
1158  U1068   132733       1            1               0
1159  U1068   132594       1            1               1
1160  U1068   132660       0            0               0

[1161 rows x 5 columns]

In [ ]:
No podemos 

In [101]:
# Realizamos una primera agrupación por PlaceID y la media de los ratins 
user_rating_2 = user_rating.groupby('placeID').agg({'rating': 'mean',
                                                   'food_rating': 'mean',
                                                   'service_rating':'mean'})
user_rating_2.head()

rating  food_rating  service_rating
placeID                                     
132560     0.50         1.00        0.250000
132561     0.75         1.00        1.000000
132564     1.25         1.25        1.500000
132572     1.00         1.00        0.933333
132583     1.00         1.00        1.250000

In [131]:
#Realizamos el merge para saber los restaurantes con la media de puntuación. 
restaurants_ratings = pd.merge(restaurants, user_rating_2, how='inner', on='placeID')
#En el merge comprobamos que con la funcionalidad "how=outer" tenemos 30 restaurantes que no tienen info general. 
#Tomamos la desición de hacer el merge con inner ya que no nos interesa no tener info general sobre un restaurante. 
restaurants_ratings.head()

placeID                        name             city country  \
0   134999             Kiku Cuernavaca       Cuernavaca  Mexico   
1   132825             puesto de tacos           s.l.p.  mexico   
2   135106  El Rinc�n de San Francisco  San Luis Potosi  Mexico   
3   132613               carnitas_mata         victoria  Mexico   
4   135040    Restaurant los Compadres  San Luis Potosi  Mexico   

             alcohol smoking_area dress_code     accessibility   price  \
0  No_Alcohol_Served         none   informal  no_accessibility  medium   
1  No_Alcohol_Served         none   informal        completely     low   
2          Wine-Beer  only at bar   informal         partially  medium   
3  No_Alcohol_Served    permitted   informal        completely  medium   
4          Wine-Beer         none   informal  no_accessibility    high   

  Rambience franchise    area other_services    rating  food_rating  \
0  familiar         f  closed           none  1.600000     1.600000   
1  familiar         f    open           none  1.281250     1.343750   
2  familiar         f    open           none  1.200000     1.200000   
3  familiar         t  closed           none  1.166667     1.333333   
4  familiar         f  closed           none  0.250000     0.250000   

   service_rating  
0          1.6000  
1          0.9375  
2          1.2000  
3          1.0000  
4          0.2500

## Problema!!
Llegados a este punto nos encontramos con que no vamos a poder realizar las correlaciones, solo tenemos datos numéricos respecto al rating. 
 

In [146]:
#Filtramos los datos para conocer los restaurantes con mayor rating. 
restaurants_ratings[restaurants_ratings.rating >= 1.8]

placeID                        name             city country  \
45   135034  Michiko Restaurant Japones  San Luis Potosi  Mexico   
82   132755        La Estrella de Dimas  San Luis Potosi  Mexico   
88   134986    Restaurant Las Mananitas       Cuernavaca  Mexico   

              alcohol smoking_area dress_code     accessibility   price  \
45  No_Alcohol_Served         none   informal  no_accessibility  medium   
82  No_Alcohol_Served         none   informal         partially  medium   
88          Wine-Beer         none     formal  no_accessibility    high   

   Rambience franchise    area other_services  rating  food_rating  \
45  familiar         f  closed           none     2.0          2.0   
82  familiar         f  closed        variety     1.8          2.0   
88  familiar         f  closed           none     2.0          2.0   

    service_rating  
45             1.6  
82             1.6  
88             2.0

In [133]:
# Queremos conocer el precio más recurrente de los restaurantes con mayor rating. 
restaurants_ratings.pivot_table(index=['price'], values=['rating'], aggfunc={'rating':'mean'})

rating
price           
high    1.246129
low     1.023122
medium  1.247401

In [135]:
restaurants_ratings.pivot_table(index=['dress_code'], values=['rating'], aggfunc={'rating':'mean'})

rating
dress_code          
casual      1.184428
formal      2.000000
informal    1.174474

In [137]:
restaurants_ratings[restaurants_ratings['dress_code'] == 'formal']


placeID                      name        city country    alcohol  \
88   134986  Restaurant Las Mananitas  Cuernavaca  Mexico  Wine-Beer   

   smoking_area dress_code     accessibility price Rambience franchise  \
88         none     formal  no_accessibility  high  familiar         f   

      area other_services  rating  food_rating  service_rating  
88  closed           none     2.0          2.0             2.0

In [134]:
restaurants_ratings.pivot_table(index=['accessibility'], values=['rating'], aggfunc={'rating':'mean'})

rating
accessibility             
completely        1.119410
no_accessibility  1.190639
partially         1.412308

No podemos concluir que los restaurantes con mayor precio tienden a tener una mayor puntuación.<br>
Tampoco que los restaurantes con una accesibilidad total tienen mayores puntuaciones. <br>
En relación al código de vestimenta, los restaurantes con una código "formal" tienen la mayor puntuación. Aunque sabemos que solo hay un restaurante por lo que no podemos establecer una relación. 

## User Profile

Debido a los inconvenientes que nos hemos encontrado con los restaurantes vamos a analizar el dataset de usuarios.

In [182]:
user_profile = pd.read_csv('userprofile.csv')
user_profile.info()
user_profile.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 138 entries, 0 to 137
Data columns (total 19 columns):
userID              138 non-null object
latitude            138 non-null float64
longitude           138 non-null float64
smoker              138 non-null object
drink_level         138 non-null object
dress_preference    138 non-null object
ambience            138 non-null object
transport           138 non-null object
marital_status      138 non-null object
hijos               138 non-null object
birth_year          138 non-null int64
interest            138 non-null object
personality         138 non-null object
religion            138 non-null object
activity            138 non-null object
color               138 non-null object
weight              138 non-null int64
budget              138 non-null object
height              138 non-null float64
dtypes: float64(3), int64(2), object(14)
memory usage: 20.6+ KB


userID   latitude   longitude smoker     drink_level dress_preference  \
0  U1001  22.139997 -100.978803  false      abstemious         informal   
1  U1002  22.150087 -100.983325  false      abstemious         informal   
2  U1003  22.119847 -100.946527  false  social drinker           formal   
3  U1004  18.867000  -99.183000  false      abstemious         informal   
4  U1005  22.183477 -100.959891  false      abstemious    no preference   

  ambience transport marital_status        hijos  birth_year    interest  \
0   family   on foot         single  independent        1989     variety   
1   family    public         single  independent        1990  technology   
2   family    public         single  independent        1989        none   
3   family    public         single  independent        1940     variety   
4   family    public         single  independent        1992        none   

           personality  religion      activity  color  weight  budget  height  
0    thrifty-protector      none       student  black      69  medium    1.77  
1  hunter-ostentatious  Catholic       student    red      40     low    1.87  
2          hard-worker  Catholic       student   blue      60     low    1.69  
3          hard-worker      none  professional  green      44  medium    1.53  
4    thrifty-protector  Catholic       student  black      65  medium    1.69

In [183]:
#Eliminamos columnas
user_profile = user_profile.drop(user_profile.columns[[1,2]], axis=1)

In [185]:
# Queremos conocer la puntuación media que ha otorgado cada usuario al total de restaurantes. 
user_rating_consumer= user_rating.groupby('userID').agg({'rating':'mean',
                                                           'food_rating':'mean',
                                                           'service_rating':'mean'})

user_rating_consumer.head()

rating  food_rating  service_rating
userID                                       
U1001   1.111111     1.222222        1.222222
U1002   1.400000     1.400000        1.000000
U1003   1.615385     1.692308        1.461538
U1004   1.875000     1.875000        1.750000
U1005   1.333333     1.444444        1.000000

In [187]:
#Realizamos una fusión de ambos dataframes. 
users = pd.merge(user_profile, user_rating_consumer, on='userID')
users.head()

userID smoker     drink_level dress_preference ambience transport  \
0  U1001  false      abstemious         informal   family   on foot   
1  U1002  false      abstemious         informal   family    public   
2  U1003  false  social drinker           formal   family    public   
3  U1004  false      abstemious         informal   family    public   
4  U1005  false      abstemious    no preference   family    public   

  marital_status        hijos  birth_year    interest          personality  \
0         single  independent        1989     variety    thrifty-protector   
1         single  independent        1990  technology  hunter-ostentatious   
2         single  independent        1989        none          hard-worker   
3         single  independent        1940     variety          hard-worker   
4         single  independent        1992        none    thrifty-protector   

   religion      activity  color  weight  budget  height    rating  \
0      none       student  black      69  medium    1.77  1.111111   
1  Catholic       student    red      40     low    1.87  1.400000   
2  Catholic       student   blue      60     low    1.69  1.615385   
3      none  professional  green      44  medium    1.53  1.875000   
4  Catholic       student  black      65  medium    1.69  1.333333   

   food_rating  service_rating  
0     1.222222        1.222222  
1     1.400000        1.000000  
2     1.692308        1.461538  
3     1.875000        1.750000  
4     1.444444        1.000000

In [188]:
#Comprobamos la correlación que existen entre las columnas numéricas
users.corr()

birth_year    weight    height    rating  food_rating  \
birth_year        1.000000  0.144163  0.243849 -0.177498    -0.199694   
weight            0.144163  1.000000  0.535570 -0.002592     0.031705   
height            0.243849  0.535570  1.000000  0.064810     0.035359   
rating           -0.177498 -0.002592  0.064810  1.000000     0.872063   
food_rating      -0.199694  0.031705  0.035359  0.872063     1.000000   
service_rating   -0.213071  0.060803  0.092694  0.840132     0.831406   

                service_rating  
birth_year           -0.213071  
weight                0.060803  
height                0.092694  
rating                0.840132  
food_rating           0.831406  
service_rating        1.000000

No existen correlaciones. 